### PR TITLE
docs: data types

### PR DESCRIPTION
**Pull request by chinathaip**

### CODE REVIEW

File: docs/data types.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
+---
+sidebar_position: 2
+---
+# Data Types
+
+*Written By: Watsamon Phongwanit*
+
+Code example can be run on: [https://runjs.co/](https://runjs.co/)
```
Comment (Lxkas) [inline docs/data types.md]: Could be
You can run the code examples on:

File: docs/data types.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+Object properties are written as name: value pairs, separated by commas.
+
+Example:
+
+```jsx
+const sandwich = {
+		name: "Ham Cheese Sandwich",
+		ingredients: ["bread", "ham", "cheese"],
+		steps: [......],
+		preptime: ....
```
Comment (Lxkas) [inline docs/data types.md]: These can be mistaken for spread operators? Maybe add a comment to clarify?

Comment (chinathaip) [reply docs/data types.md]: wdym 'spread' ?

File: docs/data types.md (hunk 1 above)
Comment (chinathaip) [reply docs/data types.md]: okie

File: docs/data types.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    console.log(book.author)
+```
+</p>    
+</details>
+
+## The typeof Operator
+
+You can use the JavaScript typeof operator to find the type of a JavaScript variable.
+
+The typeof operator returns the type of a variable or an expression:
```
Comment (Lxkas) [inline docs/data types.md]: it returns a string containing the type, not an actual type

Comment (chinathaip) [reply docs/data types.md]: okie

**Review by Lxkas — CHANGES_REQUESTED** [review]
Overall, good.
Nothing much else to note on :3

File: docs/data types.md (hunk 2 above)
Comment (Lxkas) [reply docs/data types.md]: > wdym 'spread' ?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

**Comment by chinathaip** [comment]
@Lxkas  look again pls <3

File: docs/data types.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+cars[0] // Saab
+cars[1] // Volvo
+cars[2] // BMW
+```
+
+Further explanations are available in the ***Arrays*** chapter!
+
+### Objects
+
+Consider a recipe for a sandwich. It has different parts: "ingredients," "steps," and "prep time." In JavaScript, you can create an object to hold all these details in a structured way.
```
Comment (Lxkas) [inline docs/data types.md]: Where did prep time go XD

Comment (chinathaip) [reply docs/data types.md]: Shhhh!!

**Review by Lxkas — APPROVED** [review]

**Pull request merged**